web/packages/build/jest/jest-environment-patched-jsdom.js

```diff
@@ -0,0 +1,27 @@
+import { TextEncoder, TextDecoder } from 'node:util';
+
+import JSDOMEnvironment, * as jestEnvironmentJSDOM from 'jest-environment-jsdom';
+
+// When using jest-environment-jsdom, TextEncoder and TextDecoder are not defined. This poses a
+// problem when writing tests for code which uses TextEncoder and TextDecoder directly or that
+// imports libraries which depend on those globals (for example whatwg-url).
+//
+// It's unclear if that's a problem with Jest or JSDOM itself, see
+// https://github.com/jsdom/jsdom/issues/2524#issuecomment-902027138
+//
+// In the meantime, we're just going to "polyfill" those classes from Node ourselves.
+//
+// PatchedJSDOMEnvironment is taken from:
+// https://github.com/jsdom/jsdom/issues/2524#issuecomment-1480930523
+// https://github.com/jsdom/jsdom/issues/2524#issuecomment-1542294847
+export default class PatchedJSDOMEnvironment extends JSDOMEnvironment {
+  constructor(...args) {
+    const { global } = super(...args);
+    if (!global.TextEncoder) global.TextEncoder = TextEncoder;
+    if (!global.TextDecoder) global.TextDecoder = TextDecoder;
+  }
+}
+export const TestEnvironment =
+  jestEnvironmentJSDOM.TestEnvironment === JSDOMEnvironment
+    ? PatchedJSDOMEnvironment
+    : jestEnvironmentJSDOM.TestEnvironment;
```

Comment:
Why do we need this rather than just 
```suggestion
export const TestEnvironment = PatchedJSDOMEnvironment
```

Reply:
I assumed it's some arcane incantation that Jest needs since the person who wrote the original said "don't forget that test environment export" https://github.com/jsdom/jsdom/issues/2524#issuecomment-1542538063

[The official example](https://jestjs.io/docs/configuration#testenvironment-string) doesn't even export `TestEnvironment` in the first place and it also approaches importing the original env slightly different. I simplified both imports and exports but I didn't completely remove the `TestEnvironment` export.